internal/nix/nixprofile/profile.go

```diff
@@ -26,18 +28,62 @@ func ProfileListItems(
 	profileDir string,
 ) (map[string]*NixProfileListItem, error) {
 
-	lines, err := nix.ProfileList(writer, profileDir)
+	version, err := nix.Version()
```

Comment:
Don't return error here. That puts us at a risk of failing even when we would do correct thing.

Reply:
(e.g. what if in some past version `nix --version` returned something weird, we're better off just trying with `useJSON = false`

Reply:
not sure I follow. What do you mean "failing even when we do correct thing?"

In general, I'd prefer erroring here if we fail to get the nix version properly. Otherwise, we'll error below with a more obscure error...

Reply:
If we fail to read the version and ignore the error, we can continue doing what we were doing before. Otherwise we'll have introduced a new error and potentially break old nix installations that were already working.

Reply:
I'd just try running `nix profile list --json` and if it fails try `nix profile list` (without the flag). Then you don't need to do the version comparison stuff. Example with old Nix:

```
$ nix profile list --json
error: unrecognised flag '--json'
Try 'nix --help' for more information.
$ nix profile list       
0 flake:nixpkgs#legacyPackages.aarch64-darwin.fuse github:NixOS/nixpkgs/d0f2758381caca8b4fb4a6cac61721cc9de06bd9#legacyPackages.aarch64-darwin.fuse /nix/store/8pqmbsp8zkq8656k5yaravbaxixaaa3c-macfuse-stubs-4.4.1
```

Reply:
Here's a concrete example:

Imagine that in nix v2.1.0 running `nix --version` returns `2.1.0` without the `nix (Nix)` prefix. Then this function would return an error and we would stop execution. But in this case, `version` would be `""` which would currently set `useJSON` to false. So it's a case where ignoring the error would cause correct execution but returning an error would cause execution to break. 

We don't want to break any old versions of nix, so I think just assuming useJSON is false if we can't parse the version is the correct way to go.